Orange/misc/server_embedder.py

```diff
@@ -111,25 +118,25 @@ def embedd_data(
         EmbeddingCancelledException:
             If cancelled attribute is set to True (default=False).
         """
-        # if there is less items than 10 connection error should be raised
-        # earlier
+        # if there is less items than 10 connection error should be raised earlier
         self.max_errors = min(len(data) * self.MAX_REPEATS, 10)
 
         loop = asyncio.new_event_loop()
         asyncio.set_event_loop(loop)
         try:
             embeddings = asyncio.get_event_loop().run_until_complete(
-                self.embedd_batch(data, processed_callback)
+                self.embedd_batch(data, processed_callback, callback)
             )
-        except Exception:
+        finally:
             loop.close()
-            raise
 
-        loop.close()
         return embeddings
 
     async def embedd_batch(
-            self, data: List[Any], proc_callback: Callable[[bool], None] = None
+        self,
+        data: List[Any],
+        processed_calback: Optional[Callable] = None,
+        callback: Callable = dummy_callback,
```

Comment:
Perhaps force using callback as a keyword argument (with `*`)? That will prevent unintentional mixups when the deprecated parameter is removed. Then you won't be able to use the default values though, but that could be handled internally. So perhaps `embedd_batch(self, data, *proc_callback, callback=dummy_callback)`, which after deprecation becomes   `embedd_batch(self, data, *, callback=dummy_callback)`.

Reply:
It is a good idea to force callback to be keyword arg. I add * after proc_callback since it has the default parameter (also before it had).

Reply:
Ha, I did not know you can also have it in the middle. :)